src/main/java/com/aws/greengrass/status/FleetStatusService.java

```diff
@@ -549,6 +557,14 @@ private DeploymentInformation getDeploymentInformation(Map<String, Object> deplo
                     .build();
             deploymentInformation.setStatusDetails(statusDetails);
         }
+        // Use unchangedRootComponents to update lastInstallationSource and lastReportedTimestamp in cloud.
+        // Only update the unchangedRootComponents list if a deployment is successful, because we should not display
+        // a failed deployment as component's last installation source.
+        if (deploymentDetails.containsKey(DEPLOYMENT_ROOT_PACKAGES_KEY_NAME)
+                && JobStatus.SUCCEEDED.toString().equals(deploymentInformation.getStatus())) {
+            deploymentInformation.setUnchangedRootComponents((List<String>) deploymentDetails
```

Comment:
nit: It might be misleading to see `unchangedRootComponents` being set to be the entire list of `rootPackages`. It might help to call out in the comments, that the filtering happens at a later stage.

Reply:
will add a comment